src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesWhenExpressions.java

```diff
@@ -0,0 +1,58 @@
+/*
+UnnecessaryParentheses
+tokens = (default)EXPR, IDENT, NUM_DOUBLE, NUM_FLOAT, NUM_INT, NUM_LONG, \
+         STRING_LITERAL, LITERAL_NULL, LITERAL_FALSE, LITERAL_TRUE, ASSIGN, \
+         BAND_ASSIGN, BOR_ASSIGN, BSR_ASSIGN, BXOR_ASSIGN, DIV_ASSIGN, \
+         MINUS_ASSIGN, MOD_ASSIGN, PLUS_ASSIGN, SL_ASSIGN, SR_ASSIGN, STAR_ASSIGN, \
+         LAMBDA, TEXT_BLOCK_LITERAL_BEGIN, LAND, LITERAL_INSTANCEOF, GT, LT, GE, \
+         LE, EQUAL, NOT_EQUAL, UNARY_MINUS, UNARY_PLUS, INC, DEC, LNOT, BNOT, \
+         POST_INC, POST_DEC
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+public class InputUnnecessaryParenthesesWhenExpressions {
+
+    void test(Object o) {
+
+      switch (o) {
+          case Integer i when (i > 0) -> {}  // violation 'Unnecessary paren.* around expression'
+          // violation below 'Unnecessary paren.* around expression'
+          case String s when (!s.isEmpty()) -> {}
```

Comment:
Since there are other comments to be fixed, why is this a violation below and not a violation above comment (meaning why isn't violation comment moved down 2 lines).

This isn't a terribly big deal, but later on you use more "violation above" comments. I just find it odd this one is a below. While yes we support this text, it seems more natural we would use above ones. I won't be overly picky about this.

Reply:
I don't pay attention to this. it just comes at random but ok to keep some consistency I moved all "below" to be "above"